build-system/tasks/check-links.js

```diff
@@ -62,7 +62,8 @@ function runLinkChecker(markdownFile) {
     baseUrl : 'file://' + path.dirname(path.resolve((markdownFile)))
   };
 
-  markdownLinkCheck(filteredMarkdown, opts, function(error, results) {
+  return markdownLinkCheck(filteredMarkdown, opts)
+  .then(function(error, results) {
```

Comment:
`results` will always be undefined here, as `error` will cause the promise to reject (and is thus not present in the fulfilled path)